### PR TITLE
add multipathing to installation procedure

### PR DESCRIPTION
**Pull request by SNiemann15**
Adding install steps for multipathing to IBM Z install. 
https://docs.google.com/document/d/1xPn6ZOYZcdOoOFFR20XwKrR-QHE801tQVNmH_ECs0uE/ 

This PR includes BZ https://bugzilla.redhat.com/show_bug.cgi?id=1921388

### CODE REVIEW

File: modules/installation-complete-user-infra.adoc [hunk 1]
```diff
@@ -130,6 +133,51 @@ command.
 +
 If the pod logs display, the Kubernetes API server can communicate with the
 cluster machines.
+ifdef::ibm-z[]
+. For an installation with FCP, additional steps are required to enable multipathing. 
+.. To enable multipathing on master nodes apply the following machine config:
```
Comment (bobfuru) [inline modules/installation-complete-user-infra.adoc:138]: s/nodes apply/nodes, apply

File: modules/installation-complete-user-infra.adoc [hunk 2]
```diff
@@ -130,6 +133,51 @@ command.
...
+  labels:
+    machineconfiguration.openshift.io/role: "master"
+  name: 99-master-kargs-mpath
+spec:
+  kernelArguments:
+    - 'rd.multipath=default'
+    - 'root=/dev/disk/by-label/dm-mpath-root'
+--
+
+.. To enable multipathing on worker nodes apply the following machine config:
```
Comment (bobfuru) [inline modules/installation-complete-user-infra.adoc:154]: s/nodes apply/nodes, apply
s/the following machine config:/the following machine config file: ?

File: modules/installation-complete-user-infra.adoc (hunk 1 above)
Comment (bobfuru) [reply modules/installation-complete-user-infra.adoc:138]: s/the following machine config:/the following machine config file: ?

File: modules/installation-complete-user-infra.adoc [hunk 3]
```diff
@@ -130,6 +133,51 @@ command.
...
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: "worker"
+  name: 99-master-kargs-mpath
+spec:
+  kernelArguments:
+    - 'rd.multipath=default'
+    - 'root=/dev/disk/by-label/dm-mpath-root'
+--
+.. All the worker nodes are restarted. To monitor the process, use the following command:
```
Comment (bobfuru) [inline modules/installation-complete-user-infra.adoc:169]: s/use/enter

File: modules/installation-complete-user-infra.adoc [hunk 4]
```diff
@@ -130,6 +133,51 @@ command.
...
+.. All the worker nodes are restarted. To monitor the process, use the following command: 
++
+[source,terminal]
+----
+$ oc get nodes -w
+----
++
+[NOTE]
+====
+ If you have additional machine types like infrastructure nodes, repeat the process for these types.
```
Comment (bobfuru) [inline modules/installation-complete-user-infra.adoc:178]: **! like** _conj_
Do not use to mean “such as.”

File: modules/installation-complete-user-infra.adoc [hunk 5]
```diff
@@ -130,6 +133,51 @@ command.
 +
 If the pod logs display, the Kubernetes API server can communicate with the
 cluster machines.
+ifdef::ibm-z[]
+. For an installation with FCP, additional steps are required to enable multipathing.
```
Comment (bobfuru) [inline modules/installation-complete-user-infra.adoc:137]: @bgilbert - This is the point of reference from the #multipath-47 Slack thread. 
Do you know if this addition should apply to the common UPI install docs as well? In other words, should we include these additional steps for installation with FCP to enable multipathing, or is it specific just to Z/Power?

**Review by bobfuru — COMMENTED** [review]
Hi, @SNiemann15 - A couple of minor nitpicks and I left a question for Benjamin to see if the multipathing steps should also be applied to metal and vsphere UPI installation docs as well. Thanks!

File: modules/installation-complete-user-infra.adoc (hunk 5 above)
Comment (bgilbert) [reply modules/installation-complete-user-infra.adoc:143]: It sounds as though there's now a consensus in the Slack channel about how to handle this.  I'm not a multipath SME, actually, so I'll defer to the consensus.

File: modules/installation-ibm-z-user-infra-machines-iso.adoc [hunk 6]
```diff
@@ -49,10 +49,15 @@ The rootfs image is the same for FCP and DASD.
 ... If you use static IP addresses, an empty string.
```
Comment (alkleinrh) [inline modules/installation-ibm-z-user-infra-machines-iso.adoc:49]: this is not an empty string but "none"

File: modules/installation-network-user-infra.adoc [hunk 7]
```diff
@@ -44,7 +44,7 @@ All the {op-system-first} machines require network in `initramfs` during boot
 to fetch Ignition config from the machine config server.
 
 ifdef::ibm-z[]
-During the initial boot, the machines require an FTP server in order to
+During the initial boot, the machines require an HTTP or HTTPS server in order to
 establish a network connection to download their Ignition config files.
 
 Ensure that the machines have persistent IP addresses and host names.
```
Comment (ktania46) [inline modules/installation-network-user-infra.adoc:50]: To perform the actual MPIO fail over check, leverage the bootlist command to update the boot device list with alternate logical device names.

The following command displays a boot list and it designates the possible boot devices for when the system is booted in normal mode.
$ bootlist -m normal -o
sda
To update the boot list for Normal boot mode, update additional alternate device names that you would like to add as follows:
$ bootlist -m normal -o /dev/sdc /dev/sdd /dev/sde
sdc
sdd
sde
Next node reboot will boot from the alternate device registered in the normal boot device list, when the previous/original boot disk path is down

File: modules/installation-complete-user-infra.adoc [hunk 8]
```diff
@@ -130,6 +139,51 @@ command.
...
+[source,terminal]
+----
+$ oc get nodes -w
+----
++
+[NOTE]
+====
+If you have additional machine types such as infrastructure nodes, repeat the process for these types.   
+====
+endif::ibm-z,ibm-power[]
```
Comment (ktania46) [inline modules/installation-complete-user-infra.adoc:186]: ifeval::["{context}" == "installing-ibm-power"]

To perform the actual MPIO fail over check, leverage the bootlist command to update the boot device list with alternate logical device names.

The following command displays a boot list and it designates the possible boot devices for when the system is booted in normal mode.

[source,terminal]
----
$ bootlist -m normal -o
sda
----

To update the boot list for Normal boot mode, update additional alternate device names that you would like to add as follows:

[source,terminal]
----
$ bootlist -m normal -o /dev/sdc /dev/sdd /dev/sde
sdc
sdd
sde
----

Next node reboot will boot from the alternate device registered in the normal boot device list, when the previous/original boot disk path is down

endif::[]

ifeval::["{context}" == "installing-restricted-networks-ibm-power"]

To perform the actual MPIO fail over check, leverage the bootlist command to update the boot device list with alternate logical device names.
The following command displays a boot list and it designates the possible boot devices for when the system is booted in normal mode.

[source,terminal]
----
$ bootlist -m normal -o
sda
----
To update the boot list for Normal boot mode, update additional alternate device names that you would like to add as follows:

[source,terminal]
----
$ bootlist -m normal -o /dev/sdc /dev/sdd /dev/sde
sdc
sdd
sde
----
Next node reboot will boot from the alternate device registered in the normal boot device list, when the previous/original boot disk path is down

endif::[]


@SNiemann15

File: modules/installation-complete-user-infra.adoc [hunk 9]
```diff
@@ -130,6 +139,51 @@ command.
...
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: "worker"
+  name: 99-master-kargs-mpath
+spec:
+  kernelArguments:
+    - 'rd.multipath=default'
+    - 'root=/dev/disk/by-label/dm-mpath-root'
+--
```
Comment (ktania46) [inline modules/installation-complete-user-infra.adoc:174]: ifeval::["{context}" == "installing-ibm-power"]

To perform the actual MPIO fail over check, leverage the bootlist command to update the boot device list with alternate logical device names.

The following command displays a boot list and it designates the possible boot devices for when the system is booted in normal mode.

```
$ bootlist -m normal -o
sda
```

To update the boot list for Normal boot mode, update additional alternate device names that you would like to add as follows:

```
$ bootlist -m normal -o /dev/sdc /dev/sdd /dev/sde
sdc
sdd
sde
```

Next node reboot will boot from the alternate device registered in the normal boot device list, when the previous/original boot disk path is down

endif::[]

ifeval::["{context}" == "installing-restricted-networks-ibm-power"]

To perform the actual MPIO fail over check, leverage the bootlist command to update the boot device list with alternate logical device names.
The following command displays a boot list and it designates the possible boot devices for when the system is booted in normal mode.

```
$ bootlist -m normal -o
sda
```
To update the boot list for Normal boot mode, update additional alternate device names that you would like to add as follows:

```
$ bootlist -m normal -o /dev/sdc /dev/sdd /dev/sde
sdc
sdd
sde
```
Next node reboot will boot from the alternate device registered in the normal boot device list, when the previous/original boot disk path is down

endif::[]


@SNiemann15

Comment (manojnkumar) [reply modules/installation-complete-user-infra.adoc:174]: This statement can be removed:

```
To perform the actual MPIO failover check, leverage the bootlist command to update the boot device list with alternate logical device names.
```

Also this section seems to have the same text twice.

**Comment by SNiemann15** [comment]
@ktania46 @manojnkumar I've added the Power content please have a look.

File: modules/installation-complete-user-infra.adoc [hunk 10]
```diff
@@ -130,6 +140,73 @@ command.
...
++
+[source,terminal]
+----
+$ bootlist -m normal -o /dev/sdc /dev/sdd /dev/sde
+sdc
+sdd
+sde
+----
++
+If the original boot disk path is down, the node will reboot from the alternate device registered in the normal boot device list.
```
Comment (bobfuru) [inline modules/installation-complete-user-infra.adoc:195]: s/will reboot/reboots

File: modules/installation-ibm-z-kvm-user-infra-machines-iso.adoc [hunk 11]
```diff
@@ -16,7 +16,7 @@ Before you install a cluster on IBM Z infrastructure that you provision, you mus
 
 . Install the KVM/QEMU hypervisor on the {op-system-base} KVM host.
 
-. Obtain the {op-system-base} QEMU copy-on-write (QCOW2) disk image file from the link:https://access.redhat.com/downloads/content/290[Product Downloads] page on the Red Hat customer portal or the link:https://mirror.openshift.com/pub/openshift-v4/s390x/dependencies/rhcos/4.2/latest/[{op-system} image mirror] page.
+. Obtain the {op-system-base} QEMU copy-on-write (QCOW2) disk image file from the link:https://access.redhat.com/downloads/content/290[Product Downloads] page on the Red Hat customer portal or the link:https://mirror.openshift.com/pub/openshift-v4/s390x/dependencies/rhcos/latest/[{op-system} image mirror] page.
```
Comment (bobfuru) [inline modules/installation-ibm-z-kvm-user-infra-machines-iso.adoc:19]: s/Red Hat customer portal or the/Red Hat Customer Portal or from the

File: modules/installation-ibm-z-user-infra-machines-iso.adoc [hunk 12]
```diff
@@ -36,23 +36,25 @@ The rootfs image is the same for FCP and DASD.
...
 ... The netmask.
 ... The machine host and domain name in the form `hostname.domainname`. Omit this value to let {op-system} decide.
 ... The network interface name. Omit this value to let {op-system} decide.
-... If you use static IP addresses, an empty string.
+... If you use static IP addresses, specify `none`.
 ** For `coreos.inst.ignition_url=`, specify the Ignition file for the machine role. Use `bootstrap.ign`, `master.ign`, or `worker.ign`. Only HTTP and HTTPS protocols are supported.
 ** For `coreos.live.rootfs_url=`, specify the matching rootfs artifact for the kernel and initramfs you are booting. Only HTTP and HTTPS protocols are supported.
-** All other parameters can stay as they are.
+
+** For installations on DASD type disks, specify:
```
Comment (bobfuru) [inline modules/installation-ibm-z-user-infra-machines-iso.adoc:50]: s/DASD type disks/DASD-type disks

Comment (bobfuru) [inline modules/installation-ibm-z-user-infra-machines-iso.adoc:50]: For parallelization, I recommend changing to something like:

```
** For installations on DASD-type disks, complete the following tasks:
... For `coreos.inst.install_dev=`, specify `dasda`. 
... Use `rd.dasd=` to specify the DASD where {op-system} is to be installed.
... Leave all other parameters unchanged.
```

File: modules/installation-ibm-z-user-infra-machines-iso.adoc [hunk 13]
```diff
@@ -62,6 +64,40 @@ rd.znet=qeth,0.0.bdf0,0.0.bdf1,0.0.bdf2,layer2=1,portno=0 zfcp.allow_lun_scan=0
 !condev rd.dasd=0.0.3490
 ----
 
+
+** For installations on FCP type disks, specify:
```
Comment (bobfuru) [inline modules/installation-ibm-z-user-infra-machines-iso.adoc:68]: Similar to the previous comment, it doesn't make sense to have the lead-in sentence end with "specify" if there aren't parallel actions to take in the numbered list that follows. So I suggest something like:

s/specify:/complete the following steps:

```
... Use `rd.zfcp=<adapter>,<wwpn>,<lun>` to specify the FCP disk where {op-system} is to be installed. For multipathing, repeat this step for each additional path. 
... For multipathing, set the following parameter as `rd.multipath=default`
... For multipathing,  set the install device as `coreos.inst.install_dev=/dev/mapper/mpatha`
... For single-path installation, set the install device as `coreos.inst.install_dev=sda`
```

Comment (bobfuru) [inline modules/installation-ibm-z-user-infra-machines-iso.adoc:68]: s/FCP type/FCP-type

File: modules/installation-ibm-z-user-infra-machines-iso.adoc [hunk 14]
```diff
@@ -62,6 +64,40 @@ rd.znet=qeth,0.0.bdf0,0.0.bdf1,0.0.bdf2,layer2=1,portno=0 zfcp.allow_lun_scan=0
...
+... `rd.zfcp=<adapter>,<wwpn>,<lun>` specifies the FCP disk where {op-system} is to be installed. For multipathing repeat this step for each additional path. 
+... For multipathing the following parameter must be set: `rd.multipath=default`
+... For multipathing the install device is: `coreos.inst.install_dev=/dev/mapper/mpatha`
+... For single path the install device is: `coreos.inst.install_dev=sda`
++
+[NOTE]
+====
+If additional LUNs are configured with NPIV, FCP requires `zfcp.allow_lun_scan=0`.  
+====
+... All other parameters can stay as they are.
```
Comment (bobfuru) [inline modules/installation-ibm-z-user-infra-machines-iso.adoc:78]: Leave all other parameters unchanged.

File: modules/installation-ibm-z-user-infra-machines-iso.adoc [hunk 15]
```diff
@@ -62,6 +64,40 @@ rd.znet=qeth,0.0.bdf0,0.0.bdf1,0.0.bdf2,layer2=1,portno=0 zfcp.allow_lun_scan=0
...
++
+[NOTE]
+====
+Additional post-installation steps are required to fully enable multipathing. See “Completing installation on user-provisioned infrastructure” for more information.  
+====
++
+Example parameter file, `bootstrap-0.parm`, for the bootstrap machine with multipathing:
```
Comment (bobfuru) [inline modules/installation-ibm-z-user-infra-machines-iso.adoc:85]: The following is an example parameter file `bootstrap-0.parm` for the bootstrap machine with multipathing:

File: modules/installation-network-user-infra.adoc [hunk 16]
```diff
@@ -44,7 +44,7 @@ All the {op-system-first} machines require network in `initramfs` during boot
 to fetch Ignition config from the machine config server.
 
 ifdef::ibm-z[]
-During the initial boot, the machines require an FTP server in order to
+During the initial boot, the machines require an HTTP or HTTPS server in order to
```
Comment (bobfuru) [inline modules/installation-network-user-infra.adoc:47]: s/server in order to/server to

File: modules/installation-infrastructure-user-infra.adoc [hunk 17]
```diff
@@ -25,7 +25,7 @@ page before you create the supporting infrastructure for your cluster.
 
 ifdef::ibm-z[]
 . Set up static IP addresses.
-. Set up an FTP server.
+. Set up an HTTP or HTTPS server to provide ignition files to the cluster nodes.
```
Comment (bobfuru) [inline modules/installation-infrastructure-user-infra.adoc:28]: s/ignition/Ignition

**Review by bobfuru — COMMENTED** [review]
Some suggestions for style, readability, and consistency. 😄

File: modules/installation-ibm-z-kvm-user-infra-machines-iso.adoc [hunk 18]
```diff
@@ -16,7 +16,7 @@ Before you install a cluster on IBM Z infrastructure that you provision, you mus
 
 . Install the KVM/QEMU hypervisor on the {op-system-base} KVM host.
 
-. Obtain the {op-system-base} QEMU copy-on-write (QCOW2) disk image file from the link:https://access.redhat.com/downloads/content/290[Product Downloads] page on the Red Hat customer portal or the link:https://mirror.openshift.com/pub/openshift-v4/s390x/dependencies/rhcos/4.2/latest/[{op-system} image mirror] page.
+. Obtain the {op-system-base} QEMU copy-on-write (QCOW2) disk image file from the link:https://access.redhat.com/downloads/content/290[Product Downloads] page on the Red Hat customer portal or from the link:https://mirror.openshift.com/pub/openshift-v4/s390x/dependencies/rhcos/latest/[{op-system} image mirror] page.
```
Comment (bobfuru) [inline modules/installation-ibm-z-kvm-user-infra-machines-iso.adoc:19]: s/customer portal/Customer Portal

**Pull request merged**